src/TodoApp.ts

```diff
@@ -148,14 +198,29 @@ export class TodoApp extends LitElement {
   }
 
   deleteTodo(index: number) {
-    this.todos.splice(index, 1);
-    this.requestUpdate();
+    if (!this.todos[index].completed) {
+        this.pendingTasksCount--; // Reduce pending tasks count only if the task is not completed
+      }
+      this.todos.splice(index, 1);
+      this.updatePendingTasksCount();
+      this.saveTodosToLocalStorage();
+      this.requestUpdate();
   }
 
   clearAll() {
     this.todos = [];
+    this.pendingTasksCount = 0; 
+    this.saveTodosToLocalStorage();
     this.requestUpdate();
   }
+
+  updatePendingTasksCount() {
```

Comment:
all these functions are used before they are declared, fix that

Reply:
**Yeah done**